File: frontend/src/domains/Disciplina.js
import { observable } from 'mobx';
import DomainBase from './DomainBase';

class Disciplina extends DomainBase {
  @observable id;
  @observable nome;
  @observable ementa;
  @observable conteudoProgramatico;
  @observable bibliografiaBasica;
  @observable bibliografiaComplementar;
  

  static getDomainAttributes() {
    return ['id', 'nome', 'ementa', 'conteudoProgramatico', 'bibliografiaBasica', 'bibliografiaComplementar'];
  }
}

export default Disciplina;

File: backend/src/main/java/br/gov/ac/tce/licon/entities/Disciplina.java
package br.gov.ac.tce.licon.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.validation.constraints.NotNull;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;


@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Disciplina extends AbstractAuditableEntity {

	@NotNull
	@Column(name = "NOME")
	private String nome;
	
	@NotNull
	@Column(name = "EMENTA")
	private String ementa;
	
	@NotNull
	@Column(name = "CONTEUDO_PROGRAMATICO")
	private String conteudoProgramatico;

	@NotNull
	@Column(name = "BIBLIOGRAFIA_BASICA")
	private String bibliografiaBasica;

	@NotNull
	@Column(name = "BIBLIOGRAFIA_COMPLEMENTAR")
	private String bibliografiaComplementar;


}
